src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -153,7 +153,7 @@ def deactivate(self):
         if self.is_aggregator_controlled:
             deactivate_msg = {'event': 'finish'}
             self.aggregator.add_batch_finished_event(self.area.uuid, deactivate_msg)
-        else:
+        elif self.redis_com.is_enabled:
```

Comment:
Nit:
```suggestion
        elif self.redis_communicator.is_enabled:
```
(Just for improved clarity)

Reply:
Fixing, will make it private too